libchewing make check compilation error if running ./configure with --disable-static option

I tested with latest git repo, with --disable-static options make check is failing.
How to reproduce :
1)  ./configure --disable-static 
2) make
3) make check

Fail logs :

make[2]: Entering directory /root/libchewing-master/test' CC performance.o CCLD performance CC testchewing.o CCLD testchewing ../test/.libs/libtesthelper.a(testhelper.o): In functioninternal_ok_buffer':
/root/libchewing-master/test/testhelper.c:302: undefined reference to ueStrLen' ../test/.libs/libtesthelper.a(testhelper.o): In functioninternal_ok_candidate_len':
/root/libchewing-master/test/testhelper.c:393: undefined reference to ueStrLen' ../test/.libs/libtesthelper.a(testhelper.o): In functioninternal_has_userphrase':
/root/libchewing-master/test/testhelper.c:448: undefined reference to UintFromPhone' /root/libchewing-master/test/testhelper.c:451: undefined reference toUserGetPhraseFirst'
/root/libchewing-master/test/testhelper.c:452: undefined reference to UserGetPhraseNext' /root/libchewing-master/test/testhelper.c:461: undefined reference toUserGetPhraseEnd'
collect2: error: ld returned 1 exit status
make[2]: **\* [testchewing] Error 1

but with out --disable-static option make check will pass

But after using simple ./configure make check passes.
# ============================================================================
# Testsuite summary for libchewing 0.4.0
# ============================================================================
# TOTAL: 18
# PASS: 18
# SKIP: 0
# XFAIL: 0
# FAIL: 0
# XPASS: 0
# ERROR: 0
# ============================================================================

Any suggestions ..??#

--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -84,9 +84,9 @@ AM_CPPFLAGS = \
 	$(NULL)
 
 LDADD = \
-	$(top_builddir)/src/common/libcommon.la \
 	$(top_builddir)/src/porting_layer/src/libporting_layer.la \
 	$(top_builddir)/test/libtesthelper.la \
+	$(top_builddir)/src/common/libcommon.la \
 	$(NULL)
 
 AM_LDFLAGS = -static

